### PR TITLE
StridesRange

### PR DESCRIPTION
**Pull request by sorenlassen**
StridesRange is a more concise way to use StridesIterator.

Replaced all uses of traverseStrides and mapStrides.

### CODE REVIEW

File: src/Dialect/ONNX/ElementsAttr/Strides.cpp [hunk 1]
```diff
@@ -11,19 +11,19 @@
...
 #include "src/Dialect/ONNX/ElementsAttr/Arrays.hpp"
+#include "src/Dialect/ONNX/ElementsAttr/StridesRange.hpp"
 
 using namespace mlir;
 
 namespace onnx_mlir {
 
-size_t getStridesPosition(
-    ArrayRef<int64_t> indices, ArrayRef<int64_t> strides) {
+size_t getStridesPosition(ArrayRef<uint64_t> index, ArrayRef<int64_t> strides) {
```
Comment (philass) [inline src/Dialect/ONNX/ElementsAttr/Strides.cpp:20]: Why index singular over indices?

File: src/Dialect/ONNX/ElementsAttr/StridesRange.hpp.inc [hunk 2]
```diff
@@ -0,0 +1,61 @@
+//===------------------------ StridesRange.hpp.inc ------------------------===//
+//
+//===----------------------------------------------------------------------===//
+// StridesRange template implementations
+//===----------------------------------------------------------------------===//
+
+template <size_t N>
+inline auto StridesIterator<N>::operator++() -> StridesIterator<N> & {
+  ++(value.flattenedIndex);
+  for (auto axis = shape.size();;) {
```
Comment (philass) [inline src/Dialect/ONNX/ElementsAttr/StridesRange.hpp.inc:10]: Why auto? its not consistent with the rest of the codebase.

File: src/Dialect/ONNX/ElementsAttr/StridesRange.hpp.inc [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+// efficient code from the unspecialized implementation above.
+template <>
+inline auto StridesIterator<2>::operator++() -> StridesIterator<2> & {
+  ++(value.flattenedIndex);
+  uint64_t *index = value.index.data();
+  size_t pos0 = value.pos[0];
+  size_t pos1 = value.pos[1];
+  const int64_t *strides0 = strides[0].data();
+  const int64_t *strides1 = strides[1].data();
+  for (auto axis = shape.size();;) {
```
Comment (philass) [inline src/Dialect/ONNX/ElementsAttr/StridesRange.hpp.inc:42]: here too

**Review by philass — APPROVED** [review]
LGTM!

File: src/Dialect/ONNX/ElementsAttr/StridesRange.hpp.inc (hunk 2 above)
Comment (sorenlassen) [reply src/Dialect/ONNX/ElementsAttr/StridesRange.hpp.inc]: fixed

File: src/Dialect/ONNX/ElementsAttr/StridesRange.hpp.inc (hunk 3 above)
Comment (sorenlassen) [reply src/Dialect/ONNX/ElementsAttr/StridesRange.hpp.inc]: fixed

File: src/Dialect/ONNX/ElementsAttr/Strides.cpp (hunk 1 above)
Comment (sorenlassen) [reply src/Dialect/ONNX/ElementsAttr/Strides.cpp]: I use conventions from the mlir::ElementsAttr attribute interface methods.

I added a comment to make that clear.

**Pull request merged**